### PR TITLE
Enabled Profile Favoriting, Revised Opening and Log In

### PR DESCRIPTION
**Pull request by jeffkim2001**
This PR enables profile favoriting and adds minor UI fixes, such as small adjustments to opening animation and creation of default profile for guest and staff accounts.

### CODE REVIEW

**Review by PatrickKan — CHANGES_REQUESTED** [review]
Dont allow user to favorite yourself, also I noticed that the C++ filter doesnt work (possibly because of the "+" characters, use "Patrick Kan" profile to debug

**Review by PatrickKan — APPROVED** [review]
Looks good

File: HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift [hunk 1]
```diff
@@ -122,10 +122,13 @@ extension HIEventCell {
         let pointsLabel = HILabel(style: .pointsText)
         pointsView.addSubview(pointsLabel)
         pointsLabel.constrain(to: pointsView, topInset: 0, trailingInset: 0, bottomInset: 0, leadingInset: 0)
-        pointsLabel.text = "10 Points!"
+        pointsLabel.text = "\(rhs.points) Points!"
 
         let eventType = HILabel(style: .eventType)
         eventType.text = rhs.eventType
+        if rhs.eventType == "QNA" {
```
Comment (rauhul) [inline HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift:129]: this should not be special cased in the event cell it should be an extension on eventType which implements CustomStringConvertible

File: HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift [hunk 2]
```diff
@@ -124,20 +124,18 @@ extension HIEventCell {
...
-        let eventType = HILabel(style: .eventType)
-        eventType.text = rhs.eventType
-        if rhs.eventType == "QNA" {
-            eventType.text = "COMPANY Q&A"
-        }
-        eventType.textHIColor = getTagColor(for: rhs.eventType)
-        eventType.refreshForThemeChange()
+        let eventTypeLabel = HILabel(style: .eventType)
+        let eventType = HIEventType(type: rhs.eventType)
+        eventTypeLabel.text = String(describing: eventType)
```
Comment (rauhul) [inline HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift:129]: using `.description` or "\(eventType)" is preferable

Comment (jeffkim2001) [reply HackIllinois/UI/TableView/Cells/HIBubbleCell/HIEventCell.swift]: Fixed to .description

**Review by PatrickKan — APPROVED** [review]

**Pull request merged**